core/src/test/scala/unit/kafka/log/LogCleanerTest.scala

```diff
@@ -443,28 +407,29 @@ class LogCleanerTest extends JUnitSuite {
   def testSegmentGroupingWithSparseOffsets(): Unit = {
     val cleaner = makeCleaner(Int.MaxValue)
 
-    val logProps = new Properties()
-    logProps.put(LogConfig.SegmentBytesProp, 400: java.lang.Integer)
+    val logProps = createLogProps()
     logProps.put(LogConfig.IndexIntervalBytesProp, 1: java.lang.Integer)
 
     val log = makeLog(config = LogConfig.fromProps(logConfig.originals, logProps))
 
+    // 5 bytes of dummy data
+    val dummyData = "hello".getBytes
     // fill up first segment
     while (log.numberOfSegments == 1)
-      log.appendAsLeader(TestUtils.singletonRecords(value = "hello".getBytes, key = "hello".getBytes), leaderEpoch = 0)
+      log.appendAsLeader(TestUtils.singletonRecords(value = dummyData, key = dummyData), leaderEpoch = 0)
 
     // forward offset and append message to next segment at offset Int.MaxValue
-    val records = messageWithOffset("hello".getBytes, "hello".getBytes, Int.MaxValue - 1)
+    val records = messageWithOffset(dummyData, dummyData, Int.MaxValue - 1)
     log.appendAsFollower(records)
-    log.appendAsLeader(TestUtils.singletonRecords(value = "hello".getBytes, key = "hello".getBytes), leaderEpoch = 0)
+    log.appendAsLeader(TestUtils.singletonRecords(value = dummyData, key = dummyData), leaderEpoch = 0)
```

Comment:
Hmm.. It seems a bit odd that this test mixes `appendAsFollower` and `appendAsLeader`. Maybe we could replace the calls to `appendAsLeader` with `appendAsFollower` (with an explicit offset provided).